File: DataStructures/LinkedList/printNthFromEnd.py
class Node: 
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def push(self, new_data):
        new_node = Node(new_data)
        new_node.next = self.head
        self.head = new_node
        

    def printNthFromLast(self, n):
        len = 0
        temp = self.head

        while temp is not None:
            temp = temp.next
            len+= 1

        if n > len:
            return 

        for i in range(0, len -n):
            temp = temp.next
        print(temp.data)

llist = LinkedList()
llist.push(20)
llist.push(4)
llist.push(27)
llist.push(30)

llist.printNthFromLast(4)
        

File: DataStructures/LinkedList/printNthFromEnd.java
class LinkedList {
    Node head;

    class Node {
        int data;
        Node next;
        Node(int d)
        {
            data = d;
            next = null;
        }
    }

    void push(int new_data)
    {
        Node new_node = new Node(new_data);
        //This adds to the class Node
        new_node.next = head;
        head = new_node;
    }

    void printNthFromEnd(int n)
    {
        int len = 0;
        Node temp = head;
        //this creates a temp of type node = to head
        while(temp != null)
        {
            temp = temp.next;
            len++;
        }

        if(n > len){
            return;
        }

        temp = head;

        for(int i = 0; i < len -n; i ++)
        {
            temp = temp.next;
        }

        System.out.print(temp.data);

    }
    
    public static void main(String[] args)
    {
        LinkedList llist = new LinkedList();
        llist.push(20);
        llist.push(27);
        llist.push(26);
        llist.push(24);
        llist.push(23);
        llist.push(29);
        llist.push(21);

        llist.printNthFromEnd(5);
    }

}

